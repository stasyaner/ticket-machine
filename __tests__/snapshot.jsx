import {
    act,
    create,
} from "react-test-renderer";
import Index from "../pages";

// DISCLAIMER: it's a JS because it's hard to setup TS for snapshots
// and doesn't really make sense here

const alphabet = [
    "a", "b", "c", "d", "e", "f",
    "g", "h", "i", "j", "k", "l",
    "m", "n", "o", "p", "q", "r",
    "s", "t", "u", "v", "w", "x",
    "y", "z",
];

describe("snapshot testing", () => {
    it("renders service unavailable", () => {
        let tree;
        act(() => {
            tree = create(<Index alphabet={[]} strOfStations={""} />);
        });

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it("renders service unavailable w/ alphabet but w/o stations", () => {
        let tree;
        act(() => {
            tree = create(<Index alphabet={alphabet} strOfStations={""} />);
        });

        expect(tree.toJSON()).toMatchSnapshot();
    });

    it("renders letpad", () => {
        let tree;
        act(() => {
            tree = create(<Index alphabet={alphabet} strOfStations={"test string"} />);
        });

        expect(tree.toJSON()).toMatchSnapshot();
    });
});
