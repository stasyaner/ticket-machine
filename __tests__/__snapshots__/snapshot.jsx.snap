// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing renders letpad 1`] = `
<main
  className="container"
>
  <h1>
    Train Ticket Machine
  </h1>
  <h1
    className="user-input"
  >
    
  </h1>
  <table
    className="letpad-table"
  >
    <tbody>
      <tr>
        <td>
          <button
            onClick={[Function]}
          >
            a
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            b
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            c
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            d
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            e
          </button>
        </td>
        <td />
        <td
          className="cell-st-opts"
        >
          
        </td>
      </tr>
      <tr>
        <td>
          <button
            onClick={[Function]}
          >
            f
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            g
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            h
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            i
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            j
          </button>
        </td>
        <td />
        <td
          className="cell-st-opts"
        >
          
        </td>
      </tr>
      <tr>
        <td>
          <button
            onClick={[Function]}
          >
            k
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            l
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            m
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            n
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            o
          </button>
        </td>
        <td />
        <td
          className="cell-st-opts"
        >
          
        </td>
      </tr>
      <tr>
        <td>
          <button
            onClick={[Function]}
          >
            p
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            q
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            r
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            s
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            t
          </button>
        </td>
        <td />
        <td
          className="cell-st-opts"
        >
          
        </td>
      </tr>
      <tr>
        <td>
          <button
            onClick={[Function]}
          >
            u
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            v
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            w
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            x
          </button>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            y
          </button>
        </td>
        <td />
        <td
          className="cell-st-opts"
        >
          
        </td>
      </tr>
      <tr>
        <td>
          <button
            onClick={[Function]}
          >
            z
          </button>
        </td>
        <td
          colSpan={4}
        >
          <span
            className="not-available"
          >
            space
          </span>
        </td>
        <td>
          <button
            onClick={[Function]}
          >
            del
          </button>
        </td>
        <td
          className="cell-st-opts"
        >
          
        </td>
      </tr>
    </tbody>
  </table>
</main>
`;

exports[`snapshot testing renders service unavailable 1`] = `
<main
  className="container"
>
  <h1>
    Train Ticket Machine
  </h1>
  <h1>
    Service unavailable. Please try again in a minute or report to the staff.
  </h1>
</main>
`;

exports[`snapshot testing renders service unavailable w/ alphabet but w/o stations 1`] = `
<main
  className="container"
>
  <h1>
    Train Ticket Machine
  </h1>
  <h1>
    Service unavailable. Please try again in a minute or report to the staff.
  </h1>
</main>
`;
